launcher/FileSystem.cpp

```diff
@@ -215,6 +307,275 @@ bool copy::operator()(const QString& offset, bool dryRun)
+        
+        if ((!m_recursive) || !fs::is_directory(StringUtils::toStdString(src))) {
+            if (m_debug)
+                qDebug() << "linking single file or dir:" << src << "to" << dst;
+            link_file(src, "");
+        } else {
+            if (m_debug)
+                qDebug() << "linking recursivly:" << src << "to" << dst << "max_depth:" << m_max_depth;
+            QDir src_dir(src);
+            QDirIterator source_it(src, QDir::Filter::Files | QDir::Filter::Hidden, QDirIterator::Subdirectories);
+
+            QStringList linkedPaths;
+
+            while (source_it.hasNext()) {
+                auto src_path = source_it.next();
+                auto relative_path = src_dir.relativeFilePath(src_path);
+
+                if (m_max_depth >= 0 && PathDepth(relative_path) > m_max_depth){
+                    relative_path = PathTruncate(relative_path, m_max_depth);
+                    src_path = src_dir.filePath(relative_path);
+                    if (linkedPaths.contains(src_path)) {
+                        continue;
+                    }
+                }
+
+                linkedPaths.append(src_path);
+
+                link_file(src_path, relative_path);
+            }
+        }
+    }  
+}
+
+bool create_link::make_links()
+{   
+    for (auto link : m_links_to_make) {
+
+        QString src_path = link.src;
+        QString dst_path = link.dst;
+
+        ensureFilePathExists(dst_path);
+        if (m_useHardLinks) {
+            if (m_debug)
+                qDebug() << "making hard link:" << src_path << "to" << dst_path;
+            fs::create_hard_link(StringUtils::toStdString(src_path), StringUtils::toStdString(dst_path), m_os_err);
+        } else if (fs::is_directory(StringUtils::toStdString(src_path))) {
+            if (m_debug)
+                qDebug() << "making directory_symlink:" << src_path << "to" << dst_path;
+            fs::create_directory_symlink(StringUtils::toStdString(src_path), StringUtils::toStdString(dst_path), m_os_err);
+        } else {
+            if (m_debug)
+                qDebug() << "making symlink:" << src_path << "to" << dst_path;
+            fs::create_symlink(StringUtils::toStdString(src_path), StringUtils::toStdString(dst_path), m_os_err);
+        }
+           
+
+        if (m_os_err) {
+            qWarning() << "Failed to link files:" << QString::fromStdString(m_os_err.message());
+            qDebug() << "Source file:" << src_path;
+            qDebug() << "Destination file:" << dst_path;
+            qDebug() << "Error catagory:" << m_os_err.category().name();
+            qDebug() << "Error code:" << m_os_err.value();
+            emit linkFailed(src_path, dst_path, QString::fromStdString(m_os_err.message()), m_os_err.value());
+        } else {
+            m_linked++;
+            emit fileLinked(src_path, dst_path);
+        }
+        if (m_os_err) return false;
+    }
+    return true;
+}
+
+void create_link::runPrivlaged(const QString& offset)
```

Comment:
```suggestion
void create_link::runPrivileged(const QString& offset)
```

probably needs a good `:%s/rivlaged/rivileged/g` on the file and header :P